src/blob/index/simple.rs

```diff
@@ -101,8 +101,8 @@ impl FileIndexTrait for SimpleFileIndex {
             .map(|headers| (headers, self.header.records_count))
     }
 
-    async fn get_any(&self, key: &[u8]) -> Result<Option<RecordHeader>> {
-        Self::binary_search(&self.file, key, &self.header)
+    async fn get_any(&self, key: &K) -> Result<Option<RecordHeader>> {
+        Self::binary_search(&self.file, &key.to_vec(), &self.header)
```

Comment:
`binary_search` definitely contains key comparison logic, so I guess it needs to be updated too

Reply:
Added key trait usage to this binary_search